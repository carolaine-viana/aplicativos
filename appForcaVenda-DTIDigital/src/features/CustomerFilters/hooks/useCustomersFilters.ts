import { useEffect, useState } from 'react';
import { IVisitCustomer, TPredefinedFilter } from '@shared/types';
import { ICustomerReadDto } from '@shared/models/Customer';
import { filterCustomers, searchingCustomers } from '../customersFiltersUtils';

export const useCustomersFilters = (
  customers: ICustomerReadDto[] | IVisitCustomer[] | undefined | null,
  selectedFilters: TPredefinedFilter[],
  search: string
) => {
  const [filteredData, setFilteredData] = useState<ICustomerReadDto[]>(customers);
  const [loadingFilter, setLoadingFilter] = useState(false);

  useEffect(() => {
    setLoadingFilter(true);
    const promiseFilter = (): Promise<ICustomerReadDto[]> =>
      new Promise((res) => {
        setTimeout(() => {
          if (customers) {
            const filtered = filterCustomers(selectedFilters, customers);
            const dataResult = searchingCustomers(search, filtered);
            res(dataResult);
          }
        }, 500);
      });

    const filterData = async () => {
      const data = await promiseFilter();
      setFilteredData(data);
      setLoadingFilter(false);
    };

    filterData();
  }, [search, customers, selectedFilters]);

  return {
    filteredDataSource: filteredData,
    loadingFilter,
  };
};
