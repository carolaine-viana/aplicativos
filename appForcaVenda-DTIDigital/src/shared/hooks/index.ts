export { default as useGeoLocation } from './useGeoLocation';
export * from './useNavigationFocus';
export * from './useLoadDados';
export * from './useRoleInfo';
export * from './useTimecourse';
export * from './usePromise';
export * from './useRedux';
export * from './useSearch';
export * from './useFeatureFlag';
export * from './useCombineRoleFlags';
