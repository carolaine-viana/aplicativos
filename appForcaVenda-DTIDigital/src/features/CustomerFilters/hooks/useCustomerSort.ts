import { useEffect, useState } from 'react';
import { TAlphabeticalSort } from 'shared/types';
import { ICustomerReadDto } from '@shared/models/Customer';
import { sortByName, sortBySchedule, TScheduleSortMode } from '../customersFiltersUtils';

export const useCustomerSort = (
  customers: (ICustomerReadDto & Partial<{ schedule: Date }>)[] | undefined,
  sortMode: TAlphabeticalSort | TScheduleSortMode,
  sortSchedule = false
) => {
  const [sortedData, setsortedData] = useState<ICustomerReadDto[]>(customers);
  const [loadingSort, setLoadingSort] = useState(false);

  useEffect(() => {
    if (customers) {
      setLoadingSort(true);
      const promiseSort = (): Promise<ICustomerReadDto[]> =>
        new Promise((res) => {
          setTimeout(() => {
            let sorted;
            if (sortSchedule) {
              sorted = sortBySchedule(customers, sortMode);
            } else {
              sorted = sortByName(customers, sortMode as TAlphabeticalSort);
            }
            res(sorted);
          }, 500);
        });
      const load = async () => {
        const data = await promiseSort();
        setsortedData(data);
        setLoadingSort(false);
      };
      load();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [customers, sortMode]);

  return {
    sortedData,
    loadingSort,
  };
};
