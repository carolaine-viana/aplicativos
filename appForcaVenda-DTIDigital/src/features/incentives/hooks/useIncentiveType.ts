import { useCallback } from 'react';
import { useRedux } from 'shared/hooks';
import { TIncentive } from 'shared/types';
import { setIncentiveType as setType, IIncentivesState } from '../redux/incentivesSlice';

export const useIncentiveType = () => {
  const { dispatch, sliceState } = useRedux<IIncentivesState>('incentives');

  const setIncentiveTypeMemoized = useCallback(
    (incentiveType: TIncentive) => dispatch(setType(incentiveType)),
    [dispatch]
  );

  return {
    intentiveType: sliceState.incentiveType,
    setIncentiveType: setIncentiveTypeMemoized,
  };
};
