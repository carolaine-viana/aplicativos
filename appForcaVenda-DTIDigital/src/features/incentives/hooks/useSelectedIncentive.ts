import { useCallback } from 'react';
import { useRedux } from 'shared/hooks';
import { IIncentive } from 'shared/types';
import { setSelectedIncentive as setIncentive, IIncentivesState } from '../redux/incentivesSlice';

export const useSelectedIncentive = () => {
  const { dispatch, sliceState } = useRedux<IIncentivesState>('incentives');

  const setIncentiveMemoized = useCallback(
    (incentive: Partial<IIncentive> | null) => dispatch(setIncentive(incentive)),
    [dispatch]
  );

  return {
    selectedIncentive: sliceState.selectedIncentive,
    setSelectedIncentive: setIncentiveMemoized,
  };
};
