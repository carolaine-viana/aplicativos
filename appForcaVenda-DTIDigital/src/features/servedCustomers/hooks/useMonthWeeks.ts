import { useCallback } from 'react';
import { useRedux } from 'shared/hooks';
import {
  setMonthcWeeksFilters as setWeeks,
  IServedCustomersState,
} from '../redux/servedCustomersSlice';

type TPyload = {
  idx: number;
  weeks: IServedCustomersState['selectedWeek'][];
};
export const useMonthWeeks = () => {
  const { dispatch, sliceState } = useRedux<IServedCustomersState>('servedCustomers');

  const setMonthWeeksMemoized = useCallback(
    (payload: TPyload) =>
      dispatch(
        setWeeks({
          idx: payload.idx,
          weeks: payload.weeks,
        })
      ),
    [dispatch]
  );

  return {
    monthsWeeks: sliceState.monthsWeeks,
    setMonthcWeeksFilters: setMonthWeeksMemoized,
  };
};
