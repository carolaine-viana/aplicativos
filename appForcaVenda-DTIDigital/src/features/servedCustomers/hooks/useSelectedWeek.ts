import { useCallback } from 'react';
import { useRedux } from 'shared/hooks';
import { IWeekFormats } from 'shared/types';
import { setSelectedWeek as setWeek, IServedCustomersState } from '../redux/servedCustomersSlice';

export const useSelectedWeek = () => {
  const { dispatch, sliceState } = useRedux<IServedCustomersState>('servedCustomers');

  const setSelectedWeekTypeMemoized = useCallback(
    (week: IWeekFormats & { pureDate: string }) => dispatch(setWeek(week)),
    [dispatch]
  );

  return {
    selectedWeek: sliceState.selectedWeek,
    setSelectedWeek: setSelectedWeekTypeMemoized,
  };
};
