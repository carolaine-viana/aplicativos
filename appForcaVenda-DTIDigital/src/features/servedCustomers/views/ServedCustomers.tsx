import React, { useCallback } from 'react';
import { FlatList } from 'react-native';
import { useTheme } from 'styled-components';
import { useNavigation } from '@react-navigation/native';

import { IFocusSupplierCustomer } from 'shared/types';
import { SearchHeader, CustomStatusBar, CardPaper, FallbackWrapper } from '@shared/components';
import RoleWrapper from 'shared/components/RoleWrapper/RoleWrapper';
import { usePromise, useRedux } from 'shared/hooks';

import { isArrayEmpty } from 'shared/utils';
import { SuppliersContainer } from './ServedCustomersSC';
import ServedCustomersMonthFilter from './ServedCustomersMonthFilter';
import ServedCustomersCardList from './ServedCustomersCardList';
import ServedCustomersManager from '../services/ServedCustomersManager';
import { useSelectedWeek } from '../hooks/useSelectedWeek';
import ServedCustomersSelect from './ServedCustomersSelect';
import { IServedCustomersState } from '../redux/servedCustomersSlice';

const Manager = new ServedCustomersManager();

const ServedCustomers: React.FC = () => {
  const { selectedWeek } = useSelectedWeek();
  const { sliceState } = useRedux<IServedCustomersState>('servedCustomers');

  const weekParam = selectedWeek.pureDate;
  const supplierTypeParam = sliceState.supplierType;

  const loadCustomers = useCallback(() => {
    if (weekParam) {
      return Manager.getFocusSupplierWeek(weekParam, supplierTypeParam);
    }
    return null;
  }, [weekParam, supplierTypeParam]);

  const { data: customers, ...promiseStatus } = usePromise<IFocusSupplierCustomer[]>(loadCustomers);

  const navigation = useNavigation();
  const theme = useTheme();

  return (
    <>
      <CustomStatusBar backgroundColor={theme.palette.blue50} barStyle="dark-content" />

      <SearchHeader
        onClickArrow={() => navigation.goBack()}
        headerProps={{
          title: 'Clientes Atendidos',
        }}
      />
      <SuppliersContainer>
        <RoleWrapper renderRCAContent={() => <ServedCustomersSelect />} />

        <ServedCustomersMonthFilter />

        <FallbackWrapper
          {...promiseStatus}
          showEmptyState={promiseStatus.done && isArrayEmpty(customers)}
          loadingComponent="spinner"
          renderContent={() => (
            <FlatList
              showsVerticalScrollIndicator={false}
              data={customers}
              renderItem={({ item }) => <ServedCustomersCardList customer={item} />}
              keyExtractor={(item) => item.id}
            />
          )}
        />
      </SuppliersContainer>
      <CardPaper />
    </>
  );
};

export default ServedCustomers;
