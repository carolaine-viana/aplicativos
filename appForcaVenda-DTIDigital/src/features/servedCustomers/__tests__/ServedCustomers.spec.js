// TODO: INCLUIR ESTE TESTE QUANDO TIVER FOCO E CORINGA(PROX SPRINT)
import React from 'react';
import expect from 'expect';
import { render } from '@shared/utils/test-utils';
import ServedCustomers from '../../servedCustomers/views/ServedCustomers';
import * as incentiveHooks from '../../incentives/hooks/useIncentiveType';

jest.mock('../../../shared/hooks/useRedux');
jest.mock('../../incentives/hooks/useIncentiveType', () => ({
  useIncentiveType: jest.fn(),
}));

jest.mock('styled-components', () => {
  return {
    useTheme: jest.fn().mockReturnValue({
      palette: {
        white: '#ffffff',
      },
    }),
  };
});

// TODO: No used anymore
describe.skip('TODO: Tests ServedCustomers', () => {
  it('should be render Supliers correctly', () => {
    const mockSetIncentiveTypeFunc = jest.fn();

    incentiveHooks.useIncentiveType.mockReturnValue({
      intentiveType: 'TODOS',
      setIncentiveType: mockSetIncentiveTypeFunc,
    });

    const { toJSON } = render(<ServedCustomers />);
    expect(toJSON()).toMatchSnapshot();
  });
});
