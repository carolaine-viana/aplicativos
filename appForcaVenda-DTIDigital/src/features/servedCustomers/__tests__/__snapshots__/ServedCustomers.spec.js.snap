// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TODO: Tests ServedCustomers should be render Supliers correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "",
          "height": "px",
          "zIndex": 3,
        },
      ]
    }
    translucent={true}
  />,
  <View
    backgroundColor="default"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "height": 49,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 1.5,
            "width": 1.5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1.5,
          "width": "100%",
          "zIndex": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
            "zIndex": 2,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#072AD8",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÔÑê
        </Text>
      </View>
    </View>
    <Text
      bold={true}
      color="gray400"
      fallbackText=""
      size={18}
      style={
        Array [
          Object {
            "color": "#838896",
            "fontFamily": "Rubik-Bold",
            "fontSize": 18,
            "lineHeight": 28,
            "textTransform": "none",
          },
          Object {
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
          },
        ]
      }
      variant="header"
    >
      Clientes Atendidos
    </Text>
    <View
      style={
        Array [
          Object {
            "right": 0,
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 0,
          "height": "100%",
          "overflow": "hidden",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#D3D5DA",
              "borderBottomWidth": 1,
              "width": "90%",
            },
          ]
        }
        width="90%"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 147,
            "paddingBottom": 0,
            "paddingLeft": "50%",
            "paddingRight": "50%",
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#072AD8",
              "borderRadius": 12,
              "height": 24,
              "marginBottom": 24,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 4,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 4,
              "width": 158,
            },
          ]
        }
      >
        <Text
          bold={true}
          color="light"
          fallbackText=""
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Rubik-Bold",
                "fontSize": 14,
                "lineHeight": 17,
                "textTransform": "none",
              },
            ]
          }
          variant="body"
        >
          Semana 01/01/22
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 16,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#EAF3FF",
              "borderRadius": 17,
              "paddingBottom": 38,
              "paddingLeft": 27,
              "paddingRight": 27,
              "paddingTop": 38,
            },
          ]
        }
      >
        <Text
          color="gray800"
          fallbackText=""
          style={
            Array [
              Object {
                "color": "#19202D",
                "fontFamily": "Rubik-Medium",
                "fontSize": 14,
                "lineHeight": 17,
                "textTransform": "none",
              },
              Object {
                "fontSize": 14,
                "lineHeight": 17,
                "marginBottom": 16,
              },
            ]
          }
          variant="bodyMedium"
        >
          000.000.000-00
        </Text>
        <Text
          color="gray800"
          fallbackText=""
          style={
            Array [
              Object {
                "color": "#19202D",
                "fontFamily": "Rubik-Regular",
                "fontSize": 11,
                "lineHeight": 13,
                "textTransform": "none",
              },
              Object {
                "fontSize": 12,
                "lineHeight": 19,
                "marginBottom": 8,
              },
            ]
          }
          variant="overline"
        >
          Martins
        </Text>
        <Text
          color="black"
          fallbackText=""
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Rubik-Regular",
                "fontSize": 11,
                "lineHeight": 13,
                "textTransform": "none",
              },
              Object {
                "fontSize": 12,
                "lineHeight": 19,
                "marginBottom": 8,
              },
            ]
          }
          variant="overline"
        >
          MARTINS
        </Text>
        <Text
          color="default"
          fallbackText="-"
          style={
            Array [
              Object {
                "color": "#10151D",
                "fontFamily": "Rubik-Regular",
                "fontSize": 11,
                "lineHeight": 13,
                "textTransform": "none",
              },
              Object {
                "fontSize": 11,
                "lineHeight": 13,
              },
            ]
          }
          variant="overline"
        >
          Segmento:
           
          <Text
            color="primary"
            fallbackText="-"
            style={
              Array [
                Object {
                  "color": "#072AD8",
                  "fontFamily": "Rubik-Regular",
                  "fontSize": 11,
                  "lineHeight": 13,
                  "textTransform": "none",
                },
                Object {
                  "fontSize": 11,
                  "lineHeight": 13,
                },
              ]
            }
            variant="overline"
          >
            MARTINS
          </Text>
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": 16,
                "width": "100%",
              },
            ]
          }
        >
          <View
            color="blue100"
            style={
              Array [
                Object {
                  "borderBottomColor": "#B3D3FF",
                  "borderBottomWidth": 1,
                  "width": "100%",
                },
              ]
            }
          />
        </View>
        <Text
          color="gray800"
          fallbackText=""
          style={
            Array [
              Object {
                "color": "#19202D",
                "fontFamily": "Rubik-Medium",
                "fontSize": 14,
                "lineHeight": 17,
                "textTransform": "none",
              },
              Object {
                "fontSize": 14,
                "lineHeight": 17,
                "marginBottom": 10,
                "marginTop": 24,
              },
            ]
          }
          variant="bodyMedium"
        >
          Fornecedores
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 27,
          "height": "auto",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
  </View>,
]
`;
